synapse/lib/cell.py

```diff
@@ -904,13 +905,74 @@ async def _setAhaActive(self):
         except Exception as e: # pragma: no cover
             logger.warning(f'_setAhaActive failed: {e}')
 
+    async def addActiveCoro(self, func, iden=None):
+
+        if iden is None:
+            iden = s_common.guid()
+
+        cdef = {'func': func}
+        self.activecoros[iden] = cdef
+
+        if self.isactive:
+            await self._fireActiveCoro(iden, cdef)
+
+        return iden
+
+    async def delActiveCoro(self, iden):
+
+        cent = self.activecoros.pop(iden, None)
+        if cent is None:
+            return
+
+        task = cent.get('task')
+        if task is not None:
+            task.cancel()
+            try:
+                await task
+            except asyncio.CancelledError:
+                pass
+            except Exception as e:
+                logger.warning(f'delActiveCoro Task: {e}')
+
+    async def _fireActiveCoros(self):
+        for iden, cdef in self.activecoros.items():
+            await self._fireActiveCoro(iden, cdef)
+
+    async def _fireActiveCoro(self, iden, cdef):
+        func = cdef.get('func')
+        async def wrap():
+            while not self.isfini:
+                try:
+                    await func()
+                except asyncio.CancelledError:
+                    raise
+                except Exception as e:
+                    logger.warning(f'activeCoro Error: {func} {e}')
+                    await asyncio.sleep(1)
+
+        cdef['task'] = self.schedCoro(wrap())
+
+    async def _killActiveCoros(self):
+        for iden, cdef in self.activecoros.items():
```

Comment:
unused variable iden (_iden works)

Reply:
Actually, this is a bug!  Looks like killActiveCoros() isn't actually removing them.  Will fix.